src/rpdk/core/data/schema/provider.configuration.definition.schema.hooks.v1.json

```
@@ -0,0 +1,49 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://schema.cloudformation.us-east-1.amazonaws.com/provider.configuration.definition.schema.hooks.v1.json",
+    "title": "CloudFormation Type Provider Configuration Definition MetaSchema",
+    "description": "This schema validates a CloudFormation type provider configuration definition.",
+    "type": "object",
+    "properties": {
+        "additionalProperties": {
+            "$comment": "All properties must be expressed in the schema - arbitrary inputs are not allowed",
+            "type": "boolean",
+            "const": false
+        },
+        "deprecatedProperties": {
+            "$ref": "base.definition.schema.v1.json#/properties/deprecatedProperties"
+        },
+        "allOf": {
+            "$ref": "base.definition.schema.v1.json#/definitions/schemaArray"
+        },
+        "anyOf": {
+            "$ref": "base.definition.schema.v1.json#/definitions/schemaArray"
+        },
+        "oneOf": {
+            "$ref": "base.definition.schema.v1.json#/definitions/schemaArray"
+        },
+        "required": {
+            "$ref": "base.definition.schema.v1.json#/properties/required"
+        },
+        "description": {
+            "$comment": "A short description of the type configuration. This will be shown in the AWS CloudFormation console.",
+            "$ref": "base.definition.schema.v1.json#/properties/description"
+        },
+        "properties": {
+            "type": "object",
+            "patternProperties": {
+                "(?!CloudFormation)^[A-Za-z0-9]{1,64}$": {
+                    "$comment": "TypeConfiguration properties starting with `CloudFormation` are reserved for CloudFormation use",
```

```suggestion
                    "$comment": "HookConfiguration properties starting with `CloudFormation` are reserved for CloudFormation use",
```
(I'm not 100% sure about this one, "HookConfiguration" sounds strange)

---

Im not 100% percent sold on this also. I'd rather keep how it is to be more inline with how the configuration properties are declared in the schema